KS-20

seung-woo-ryu opened the pull request:
### 아이템 댓글
- POST 댓글 등록 (O)
- DELTE 댓글 삭제 (O)
- GET 특정 아이템에 등록된 댓글 전체 가져오기 (O)
- GET 아이템에 등록된 댓글 개수 (O)

### TestDataRunner
- 테스트용 데이터 생성 (O)

seung-woo-ryu committed:
feat: add Exception Handler about @Valid

seung-woo-ryu committed:
fix: resolve conflict

seung-woo-ryu committed:
KS-20 add: add test data class

- entity에 builder 추가
- MemberRoleRepository, MemberRepository 추가
- 테스트 데이터 추가

seung-woo-ryu committed:
fix: alter property name value to content because name was reserved word

seung-woo-ryu committed:
KS-20 style: change test data naming

- 보기 쉽게 테스트 데이터 대문자, _ 형태로 바꿈.
- 특정 테스트에 필요한 경우 뒤에 관련된 이름 붙여줌(ex:
  ITEM1_THAT_COMMENT_COUNT3

seung-woo-ryu committed:
add: add http logging config

seung-woo-ryu committed:
KS-20 fix: change json name _ to -

seung-woo-ryu committed:
KS-20 fix: 예약어라 변수명 바뀜 메서드 명도 같이 바꿈

seung-woo-ryu committed:
KS-20 add: add integration test code

seung-woo-ryu committed:
KS-20 fix: 잘못 쓰여진 import문, 필요없는 메서드 제거

seung-woo-ryu commented on src/main/java/com/kreamish/kream/comment/entity/Comment.java:
```diff
@@ -37,5 +39,13 @@ public class Comment extends BaseEntity {
     @Column
     @Length(min = 1, max = 2000)
-    private String value;
+    private String content;
```
책님 여기 원래 String value였는데 DB 예약어여서 content로 변수명 바꿨어요

seung-woo-ryu commented on src/main/java/com/kreamish/kream/common/runner/TestDataRunner.java:
```diff
@@ -0,0 +1,106 @@
+        // categoryDetail
+        DETAIL1_WITH_CATEGORY1 = saveCategoryDetail(CATEGORY1, "categoryDetail1");
+        // item
+        ITEM1_WITH_BRAND1_CATEGORY1_DETAIL1_AND_COMMENT_CNT_IS_2 = saveItem("item1", "subName1",
```
이름이 복잡해서 바꿔봐야할거 같은데 어떻게 할지 모르겠네요 ㅠㅠ
변수명 마지막에 특정 테스트와 연관되는 이름 넣는거 좋은거 같아요
COMMENT_CNT_IS_2 <- 이런거 넣어봤는데 나쁘진 않은거 같습니다

seung-woo-ryu committed:
KS-20 style: 공백 제거

seung-woo-ryu committed:
KS-20 style: 공백 제거

seung-woo-ryu commented on src/main/java/com/kreamish/kream/common/runner/TestDataRunner.java:
```diff
@@ -0,0 +1,106 @@
+@Profile("data & test")
+@Component
+@RequiredArgsConstructor
+public class TestDataRunner implements ApplicationRunner {
```
테스트 데이터 코드 KS-24보다 KS-20(현재 보고있는)에 작성된게 많아서
KS-24 폐기하고 이걸로 같이 푸쉬할게요

kangyunsik replied:
네 test코드 쪽에 작성할 수 있다면 좋을 거 같습니다

seung-woo-ryu committed:
KS-20 add: add Transactional(readOnly=true)option

kangyunsik commented on src/main/java/com/kreamish/kream/comment/controller/CommentController.java:
```diff
@@ -0,0 +1,92 @@
+
+    private final CommentFacade commentFacade;
+
+    @PostMapping("/comment")
```
comments 로 변경하는게 좋지 않을까 싶습니다 !
restful api 에서 관례적으로, 복수형을 쓰는게 좋을 것 같습니다.

kangyunsik commented on src/main/java/com/kreamish/kream/comment/controller/CommentController.java:
```diff
@@ -0,0 +1,92 @@
+            HttpStatus.OK);
+    }
+
+    @DeleteMapping("/comment/{comment-id}/member/{member-id}")
```
댓글은 crud 외에 다른 기능이 없을 듯 하여, 사용자 위주로 댓글보다 먼저 path variable 에서 사용하면 좋지 않을까 싶습니다. 추후 구매나 판매 입찰 등 사용자가 주체로 api를 호출하는 경우가 많을 듯 하여, /members/{member-id}/comments/{comments-id} 로 사용하는 것이 좋지 않을까 생각됩니다.

kangyunsik replied:
우선 복수형 변경하고, /member/member-id 부분은 request parameter 로 빼 두었다가 추후 변경하시죠 !

kangyunsik commented on src/main/java/com/kreamish/kream/member/service/MemberService.java:
```diff
@@ -0,0 +1,21 @@
+
+    private final MemberRepository memberRepository;
+
+    @Transactional(readOnly = true)
```
서비스에 Transactional 있어서, 이 부분 제거해도 될 거 같습니다 !

seung-woo-ryu committed:
KS-20 fix: modify reviewed code

- move dataRunner source package to test package
- remove pathvariable and add requestparam

seung-woo-ryu committed:
KS-20 style: url 단수형 -> 복수형

seung-woo-ryu committed:
KS-20 style: url 단수형 -> 복수형

seung-woo-ryu committed:
KS-20 style: url 단수형 -> 복수형

seung-woo-ryu committed:
KS-20 add: add @EnableJpaAuditing
